tests/ops/test_qubit_ops.py

```diff
@@ -685,6 +685,56 @@ def test_ISWAP_decomposition(self, tol):
 
         assert np.allclose(decomposed_matrix, op.matrix, atol=tol, rtol=0)
 
+    def test_toffoli_decomposition(self, tol):
+        """Tests that the decomposition of the Toffoli gate is correct"""
+        op = qml.Toffoli(wires=[0, 1, 2])
+        res = op.decomposition(op.wires)
+
+        assert len(res) == 15
+
+        mats = []
+
+        for i in reversed(res):
+            if i.wires == Wires([2]):
+                mats.append(np.kron(np.eye(4), i.matrix))
+            elif i.wires == Wires([1]):
+                mats.append(np.kron(np.eye(2), np.kron(i.matrix, np.eye(2))))
+            elif i.wires == Wires([0]):
+                mats.append(np.kron(i.matrix, np.eye(4)))
+            elif i.wires == Wires([0, 1]) and i.name == "CNOT":
```

Comment:
Why do we need the ``and`` in this conditional statement? Based on this [figure](https://github.com/PennyLaneAI/pennylane/pull/1320#issue-645193440) all two-qubit gates entering the decomposition are CNOTs. am I missing something?

Reply:
Hi! Thank you for the review @agran2018. Actually, I am a bit new to writing tests and referred to the `test_ISWAP_decomposition` function to write the test for Toffoli decomposition to maintain uniformity. In that function too `and` was given in the conditional statement even though the only two-qubit gate that was present was the CNOT. Should I make the required changes and remove the `and` condition?
Thanks!

Reply:
Thank you @tgag17 for clarifying this. @antalszava just explained that one could potentially change the decomposition, where it could be nice to have it be explicit.